src/pages/car/Car.js

```diff
@@ -51,23 +56,31 @@ const Car = () => {
         {car.price}
         /day
       </p>
-      <form onReset={handleSubmit}>
-        <label>
-          City:
-          <input type="text" name="city" onChange={handleChange} />
-        </label>
-        <label>
-          Duration:
-          <input type="text" name="duration" onChange={handleChange} />
-        </label>
-        <label>
-          Date:
-          <input type="text" name="date_reserved" onChange={handleChange} />
-        </label>
-        { car.reserved ? (<button type="button" disabled>Reserved</button>)
-          : (<button type="button" onClick={createReserve}>Reserve</button>)}
-      </form>
-      <p>{car.description}</p>
+      <Button className="nextButton" onClick={handleShow}>
+        Reserve Car
+      </Button>
+      <Modal show={show} onHide={handleClose} size="sm">
+        <Modal.Header closeButton>
+          <Modal.Title>Add Reservation</Modal.Title>
+        </Modal.Header>
+        <form onReset={handleSubmit}>
+          <label>
+            City:
+            <input type="text" name="city" onChange={handleChange} />
+          </label>
+          <label>
+            Duration:
+            <input type="text" name="duration" onChange={handleChange} />
+          </label>
+          <label>
+            Date:
+            <input type="text" name="date_reserved" onChange={handleChange} />
+          </label>
+          { car.reserved ? (<button type="button" disabled>Reserved</button>)
+            : (<button type="button" onClick={createReserve}>Reserve</button>)}
+        </form>
+        <p>{car.description}</p>
+      </Modal>
     </>
```

Comment:
Good job there the reserve action works fine, however, in the requirement, we are supposed to create a page, not a model.
The user can access the "Reserve" page from the navigation panel and also from the car details page when they click on the "Reserve" button. 
if the user reserve from the car details page username and selected car are auto-filled and if the user reserve from the navbar only the username is auto-filled.